fe1-web/src/features/wallet/screens/WalletHome.tsx

```diff
@@ -1,49 +1,117 @@
 import { useNavigation } from '@react-navigation/native';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { StyleSheet, View, ViewStyle } from 'react-native';
+import { useSelector } from 'react-redux';
 
-import { TextBlock, WideButtonView } from 'core/components';
+import { QRCode, TextBlock, WideButtonView } from 'core/components';
 import containerStyles from 'core/styles/stylesheets/containerStyles';
+import { LaoEventType } from 'features/events/objects';
+import { makeEventByTypeSelector } from 'features/events/reducer';
+import { selectCurrentLao } from 'features/lao/reducer';
+import { RollCall } from 'features/rollCall/objects';
 import STRINGS from 'resources/strings';
 
-import { WalletStore } from '../store';
+import { RollCallTokensDropDown } from '../components';
+import * as Wallet from '../objects';
+import { createDummyWalletState, clearDummyWalletState } from '../objects/DummyWallet';
+import { RollCallToken } from '../objects/RollCallToken';
 
 const styles = StyleSheet.create({
+  homeContainer: {
+    ...containerStyles.centeredXY,
+    padding: 30,
+  } as ViewStyle,
   smallPadding: {
     padding: '1rem',
   } as ViewStyle,
-  largePadding: {
-    padding: '2rem',
+  tokenSelectContainer: {
+    flexDirection: 'row',
+    marginTop: 30,
+    marginBottom: 20,
   } as ViewStyle,
 });
 
+const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
+
 /**
- * Wallet home screen
+ * Wallet UI once the wallet is synced
  */
 const WalletHome = () => {
-  // FIXME: Navigation should use a defined type here (instead of any)
+  const [tokens, setTokens] = useState<RollCallToken[]>();
+  const [selectedTokenIndex, setSelectedTokenIndex] = useState(-1);
+  const [isDebug, setIsDebug] = useState(false);
+  const rollCalls = useSelector(rollCallSelector);
+  const lao = useSelector(selectCurrentLao);
   const navigation = useNavigation<any>();
```

Comment:
It is probably a good idea to leave the `// FIXME: Navigation should use a defined type here (instead of any)` comment so that it is not forgotten when somebody takes care of defining the navigation types? :)